docs/references/debugging-sdk-tools.md

```diff
@@ -0,0 +1,82 @@
+
+
+## 4. Running the plugin at command line
+If you see an error message that looks like `Internal error: Failed to read port assignment`, you can try to run the `main.pyz` from your tool workspace using the command line to diagnose the error that's preventing the port assignment. Navigate to your tool workspace and run the following command:
+
+`python main.pyz start-sdk-tool-service ayx_plugins <tool_name>`
+A successful run will look like this:
+
+```sh
+(dev) orion.ou ~\tool_workspace [11:54AM]> python .\main.pyz start-sdk-tool-service ayx_plugins Passthrough
+ListenPort: None
+INFO: Starting shivRemoval tool with AMP Provider.
+ListenPort: 62136
+```
+
+If there are any issues with your python code, it would look something like this:
+
+```sh
+(dev) orion.ou ~\post-shiv-removal [01:08PM]> python .\main.pyz start-sdk-tool-service ayx_plugins shivRemoval
+ListenPort: None
+INFO: Starting shivRemoval tool with AMP Provider.
+EXCEPTION: name 'badcode' is not defined
+Traceback (most recent call last):
+  File ".\main.pyz", line 39, in <module>
+    start_sdk_tool_service(tool_package, tool_name, os.getenv("TOOL_SERVICE_ADDRESS"))
+  File "c:\users\orion.ou\sdks\ayx-core\ayxsdk\ayxpythonsdk\ayx_python_sdk\providers\amp_provider\__main__.py", line 48, in start_sdk_tool_service
+    plugin_class = load_plugin_class(plugins_package, tool_name)
+  File "c:\users\orion.ou\sdks\ayx-core\ayxsdk\ayxpythonsdk\ayx_python_sdk\providers\amp_provider\plugin_class_loader.py", line 62, in load_plugin_class
+    plugins_module = importlib.import_module(plugins_package)
+  File "C:\Users\orion.ou\Miniconda3\envs\dev\lib\importlib\__init__.py", line 127, in import_module
+    return _bootstrap._gcd_import(name[level:], package, level)
+  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
+  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
+  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
+  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
+  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
+  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
+  File "C:\Users\orion.ou\workspaces\post-shiv-removal\.ayx_cli.cache\dist\ayx_plugins\__init__.py", line 15, in <module>
+    from .shiv_removal import shivRemoval
+  File "C:\Users\orion.ou\workspaces\post-shiv-removal\.ayx_cli.cache\dist\ayx_plugins\shiv_removal.py", line 21, in <module>
+    badcode
+NameError: name 'badcode' is not defined
+```
+
+## 5. Simple ways to print/check values
+In your plugin code, use the `self.provider.io.info("")` method to add output to your tool that will show up in Designer. This is an easy and simple way to look at order in which your code is executing in live runs of the plugin. Using this method is a bit more convenient than using python's `println()` method since you would have to check the `PythonSDK.log` to see those print statements. 
```

Comment:
TW review:

## 5. Simple Ways to Print and Check Values
In your plugin code, use the `self.provider.io.info("")` method to add output to your tool that will show up in Designer. This is an easy and simple way to look at the order in which your code executes in live runs of the plugin. Using this method is a bit more convenient than using Python's `println()` method since you would have to check the `PythonSDK.log` to see those print statements.